pallets/marketplace/src/lib.rs

```diff
@@ -222,67 +225,15 @@ pub mod pallet {
         ) -> DispatchResultWithPostInfo {
             let caller_id = ensure_signed(origin)?;
 
-            ensure!(commission_fee <= 100, Error::<T>::InvalidCommissionFeeValue);
-            check_bounds(
-                name.len(),
-                (T::MinNameLen::get(), Error::<T>::TooShortMarketplaceName),
-                (T::MaxNameLen::get(), Error::<T>::TooLongMarketplaceName),
-            )?;
-
-            if let Some(text) = uri.as_ref() {
-                check_bounds(
-                    text.len(),
-                    (T::MinUriLen::get(), Error::<T>::TooShortUri),
-                    (T::MaxUriLen::get(), Error::<T>::TooLongUri),
-                )?;
-            }
-
-            if let Some(text) = logo_uri.as_ref() {
-                check_bounds(
-                    text.len(),
-                    (T::MinUriLen::get(), Error::<T>::TooShortLogoUri),
-                    (T::MaxUriLen::get(), Error::<T>::TooLongLogoUri),
-                )?;
-            }
-
-            if let Some(text) = description.as_ref() {
-                check_bounds(
-                    text.len(),
-                    (T::MinDescriptionLen::get(), Error::<T>::TooShortDescription),
-                    (T::MaxDescriptionLen::get(), Error::<T>::TooLongDescription),
-                )?;
-            }
-
-            // Needs to have enough money
-            let imbalance = T::Currency::withdraw(
-                &caller_id,
-                MarketplaceMintFee::<T>::get(),
-                WithdrawReasons::FEE,
-                KeepAlive,
-            )?;
-            T::FeesCollector::on_unbalanced(imbalance);
-
-            let marketplace = MarketplaceInformation::new(
+            <Self as MarketplaceTrait<T::AccountId>>::create(
```

Comment:
This is not the right way to do it. The trait is supposed to call the create extrinsic.